src/lando/api/views.py

```diff
@@ -104,3 +110,45 @@ class Form(forms.Form):
             )
 
         return JsonResponse({"errors": dict(form.errors)}, status=400)
+
+
+class CommitMapBaseView(View):
+    """CommitMap base view to be extended for bidirectional git - hg mapping."""
+
+    scm = None
+
+    def get(
+        self, request: WSGIRequest, git_repo_name: str, commit_hash: str
+    ) -> JsonResponse:
+        hash_field = f"{self.scm}_hash"
+        filters = {hash_field: commit_hash, "git_repo_name": git_repo_name}
+        commits = CommitMap.objects.filter(**filters)
+
+        if not commits.exists():
+            CommitMap.catch_up(git_repo_name)
```

Comment:
How long does this take? Is it likely to last more than a request lifetime, and if so what would happen if the client request dies? Can we leapfrog?

Maybe it's something we could delegate to a celery worker?